Sources/OTT/Services/BookmarkService.swift

```diff
@@ -29,17 +30,20 @@ internal class BookmarkService {
                 .set(method: .post)
                 .setBody(key: "ks", value: JSON(ks))
                 .setBody(key: "bookmark",
-                         value: createBookmark(eventType: eventType, position: currentTime, assetId: assetId, assetType: assetType, fileId: fileId))
+                         value: createBookmark(eventType: eventType, position: currentTime, assetId: assetId, epgId: epgId, assetType: assetType, fileId: fileId))
             return request
         } else {
             return nil
         }
     }
 
-    private static func createBookmark(eventType: String, position: Int32, assetId: String, assetType: String, fileId: String) -> JSON {
+    private static func createBookmark(eventType: String, position: Int32, assetId: String, epgId: String? = nil, assetType: String, fileId: String) -> JSON {
         var json: JSON = JSON.init(["objectType": "KalturaBookmark"])
         json["type"] = JSON(assetType)
         json["id"] = JSON(assetId)
+        if let epgId = epgId {
+            json["programId"] = JSON(epgId) // not alwayes only if epgId exists
+        }
```

Comment:
The comment is not needed, it is understood from the if statement.
```suggestion
        if let epgId = epgId {
            json["programId"] = JSON(epgId)
        }
```